### PR TITLE
fix: Handle all exceptions when creating tones for Wire's notifications

### PR DESCRIPTION
**Pull request by makingthematrix**
When we open Wire, we try to register our own custom sounds for notifications: one for new message, and one for a ping. (Right now we're doing it every time we open the app - maybe we don't have to do it so often).
In order to do that we have to find the external notifications folder, find a tone file within it, copy our sound to that file, and notice Android about it by inserting some data into the content resolver. At each step of this process something may go wrong and an exception will be thrown. We try to handle those exceptions, but it seems that at least on some phones with Android 10 one of the exceptions escapes. [You can look into Google Play Console for that](https://play.google.com/apps/publish/?account=7098984309886892484#AndroidMetricsErrorsPlace:p=com.wire&appid=4973241010395499500&appVersion=PRODUCTION&clusterName=apps/com.wire/clusters/501b2c9d&detailsAppVersion=PRODUCTION&detailsSpan=7).

This PR refactors the process, wrapping everything in functional "Tries" in Scala. This way we should be able to catch all exceptions and handle them gracefully.

#### APK
[Download build #1355](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1355/artifact/build/artifact/wire-dev-PR2633-1355.apk)
[Download build #1370](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1370/artifact/build/artifact/wire-dev-PR2633-1370.apk)

### CODE REVIEW

**Comment by makingthematrix** [comment]
build again

File: app/src/main/scala/com/waz/zclient/notifications/controllers/NotificationManagerWrapper.scala [hunk 1]
```diff
@@ -314,7 +315,16 @@ object NotificationManagerWrapper {
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
 
       addToExternalNotificationFolder(R.raw.new_message_gcm, getString(R.string.wire_notification_name))
+        .recover {
+          case ex: Exception =>
+            error(l"Failed to add ${(R.raw.new_message_gcm, getString(R.string.wire_notification_name))} to the external notification folder", ex)
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/notifications/controllers/NotificationManagerWrapper.scala]: AFAIK, these R.raw values will be converted to some integer, which changes in every build. So this error message might be not as explanatory as we want.

Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/notifications/controllers/NotificationManagerWrapper.scala]: ok, I removed the raw values from the texts.

**Comment by makingthematrix** [comment]
build again

**Pull request merged**